daemon/container_operations_unix.go

```diff
@@ -63,7 +63,11 @@ func (daemon *Daemon) getIpcContainer(container *container.Container) (*containe
 	if err != nil {
 		return nil, errors.Wrapf(err, "cannot join IPC of a non running container: %s", container.ID)
```

Comment:
> does daemon.GetContainer(containerID) actually produce an error if the container is not running?

No. This is checked later by daemon.CheckContainer()). I guess you were misled by the error message ("cannot join IPC of a non running container"), I am going to change it to:
```go
return nil, errors.Wrapf(err, "cannot join IPC of nonexistent container: %s", donorID)
```

> Testing on current docker, it looks like not;
> ...
> docker: Error response from daemon: linux spec namespaces: container 861827549edaa5c9fbd08f72bd621d1ae8f5c688d8e89d4c35d9437c4a160537 is not running.

This is the error message from the abovementioned daemon.CheckContainer().

> Paused containers are ok?
Yes.

> I also notice that the container argument is being re-used here
Oh what a mess, I should have noticed it too. Fixed.